modules/basics/src/main/java/com/opengamma/strata/basics/currency/FxRate.java

```diff
@@ -236,6 +237,23 @@ public FxRate toConventional() {
 
   //-------------------------------------------------------------------------
   /**
+   * Returns an instance with the rate updated.
+   * <p>
+   * This returns an {@code FxRate} with the same currency pair and an updated rate.
+   * For example, the operator could be used to add the forward points of an FX swap.
+   * <pre>
+   *   forwardRate = nearRate.mapRate(rate -> rate + forwardPoints);
+   * </pre>
+   *
+   * @param operator  the operator to be applied to the existing rate
+   * @return a new instance with the rate updated
+   */
+  public FxRate mapRate(DoubleUnaryOperator operator) {
```

Comment:
That is by design, as I said in the comment. If you need the rate as a double you just call FxRate.fxRate(base, counter). What’s hard about that? It’s also completely unambiguous.


> On 11 Mar 2016, at 17:45, Stephen Colebourne <notifications@github.com> wrote:
> 
> In modules/basics/src/main/java/com/opengamma/strata/basics/currency/FxRate.java:
> 
> > @@ -236,6 +237,23 @@ public FxRate toConventional() {
> >  
> >    //-------------------------------------------------------------------------
> >    /**
> > +   * Returns an instance with the rate updated.
> > +   * <p>
> > +   * This returns an {@code FxRate} with the same currency pair and an updated rate.
> > +   * For example, the operator could be used to add the forward points of an FX swap.
> > +   * <pre>
> > +   *   forwardRate = nearRate.mapRate(rate -> rate + forwardPoints);
> > +   * </pre>
> > +   *
> > +   * @param operator  the operator to be applied to the existing rate
> > +   * @return a new instance with the rate updated
> > +   */
> > +  public FxRate mapRate(DoubleUnaryOperator operator) {
> 
> I've removed the method. But really I feel we are making it too hard to access the rate.
> 
> —
> Reply to this email directly or view it on GitHub.
>